docs/source/layout.rst

```diff
@@ -0,0 +1,680 @@
+         this_column as what_we_align_to,
+         another_column as something_short,  -- Is ok
+         case
+            -- This is aligned correctly with below
+            when indented then take_care
+            else try_harder
+         end as the_general_guidance
+      -- Even here we align with the line below
+      FROM my_table
+
+   .. note::
+
+      When fixing issues with comment indentation, SQLFluff
+      will attempt to keep comments in their original position
+      but if line length concerns make this difficult, it will
+      either abandon the fix, or move *same line* comments up and
+      *before* the line they are currently on. This is in line
+      with the assumption that comments on their own line refer
+      to the elements of code which they come *before*, not *after*.
+
+
+.. _hangingindents:
+
+Hanging Indents
+^^^^^^^^^^^^^^^
+
+One approach to indenting nested elements is a layout called a
+*hanging indent*. In this layout, there is no line break before the
+first nested element, but subsequent elements are indented to
+match the line position of that first element. Two examples might be:
+
+.. code-block:: sql
+
+   -- A select statement with two hanging indents:
+   SELECT no_line_break_before_me,
+          indented_to_match_the_first,
+          1 + (a
+               + b) AS another_more_complex_example
+   FROM my_table;
+
+   -- This TSQL example is also in essence a hanging indent:
+   DECLARE @prv_qtr_1st_dt DATETIME,
+           @last_qtr INT,
+           @last_qtr_first_mn INT,
+           @last_qtr_yr INT;
+
+In some circumstances this layout can be quite neat (the
+:code:`DECLARE` statement is a good example of this), however
+once indents are nested or indentation styles are mixed it
+can rapidly become confusing (as partially shown in the first
+example). Additionally, unless the leading element of the first
+line is very short, hanging indents use much *larger indents*
+than a traditional simple indent where a line break is used before
+the first element.
+
+Hanging indents have been supported in SQLFluff up to the 1.x
+versions, however **they will no longer by supported from 2.0.0**
+onwards. This is due to the ambiguity which they bring to
+fixing poorly formatted SQL. Take the following code:
+
+.. code-block:: sql
+
+   SELECT   this_is,
+   badly_formatted, code_and,
+      not_obvious,
+         what_was,
+   intended FROM my_table
+
+Given the lack of line break between :code:`SELECT` and
+:code:`this_is`, it would appear that the user is intending
+a hanging indent, however it is also plausible that they did
+not and they just forgot to add a line break between them.
+This ambiguity is unhelpful, both for SQLFluff as a tool,
+but also for people who write SQL that there two ways of
+indenting their SQL. Given SQLFluff aims to provide consistency
+in SQL layout and remove some of the burden of needing to make
+choices like this - and that nobody is thinking of using *only*
```

Comment:
I'm totally fine to ditch hanging indents (they don't make sense beyond basic SQL examples), but not sure I follow this:

> nobody is thinking of using *only* hanging indents

Is that because you too are implying they only work for some scenarios and therefore no one uses them consistently, making it unhelpful for SQLFluff to try and apply it universally (which is what config is for)?

Reply:
I'm saying that whenever they're used - it's usually in combination with traditional indents. There aren't any projects where _traditional_ indents aren't allowed.

Reply:
I'll make that a little clearer before merging.